engine/src/main/java/org/hibernate/search/util/logging/impl/Log.java

```diff
@@ -736,4 +737,6 @@
 	@Message(id = 248, value = "WorkList should never be empty. Stacktrace below \n %s" )
 	void workListShouldNeverBeEmpty(String stackTrace);
 
+	@Message(id = 249, value = "Unsupported value type for configuration property " + Environment.ERROR_HANDLER + ": %1$s")
+	SearchException unsupportedErrorHandlerConfigurationValueType(@FormatWith(ClassObjectFormatter.class) Class<?> errorHandlerValueType);
```

Comment:
Hum, didn't expect this to cause a big discussion tbh.

Consider these other random examples from the same logger class:

* @Message(id = 46, value = "FieldCache was enabled on class %1$s but for
this type of identifier we can't extract values from the FieldCache: cache
disabled"))
* @Message(id = 125, value = "Interceptor enforces skip index operation
%2$s on instance of class %1$s")

Here "class" will be printed out twice in the rendered message.

* @Message(id = 156, value = "Cannot read %1$s field from a %2$s object :
does getter exists and is it public ?")

Here the class prefix reads odd.

* @Message(id = 134, value = "Unable to instantiate ClassBridge of type
%1$s defined on %2$s")

This one requires the calling code to call getName()

Etc. etc.

The formatter helps to avoid this sort of inconsistencies.